cli/list.go

```diff
@@ -127,14 +127,17 @@ func hasExtension(ws codersdk.Workspace) (bool, time.Duration) {
 	if ws.LatestBuild.Transition != codersdk.WorkspaceTransitionStart {
 		return false, 0
 	}
+	if ws.LatestBuild.Job.CompletedAt == nil {
+		return false, 0
+	}
 	if ws.LatestBuild.Deadline.IsZero() {
 		return false, 0
 	}
 	if ws.TTLMillis == nil {
 		return false, 0
 	}
 	ttl := time.Duration(*ws.TTLMillis) * time.Millisecond
-	delta := ws.LatestBuild.Deadline.Add(-ttl).Sub(ws.LatestBuild.CreatedAt)
+	delta := ws.LatestBuild.Deadline.Add(-ttl).Sub(*ws.LatestBuild.Job.CompletedAt)
```

Comment:
*review:* this was causing us to measure deadline extensions incorrectly, resulting in things like `8h (+3m)` if your workspace took 3 minutes to buidl.